File: 백준/Silver/1940. 주몽/주몽.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        int[] nums = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .sorted()
                .toArray();

        int left = 0;
        int right = n-1;
        int count = 0;

        while (left < right) {
            int sum = nums[left] + nums[right];

            if (sum == m) {
                count++;
                left++;
                right--;
            } else if (sum < m) {
                left++;
            } else {
                right--;
            }
        }

        System.out.println(count);
    }
}

File: 백준/Silver/1940. 주몽/주몽.kt
/**
 * p1940 주몽 (투포인터)
 */
fun main() {
    val n = readln().toInt()
    val m = readln().toInt()
    val nums = readln().split(" ").map { it.toInt() }.sorted()

    var left = 0
    var right = n-1
    var count = 0

    while (left < right) {
        val sum = nums[left] + nums[right]

        if (sum == m) {
            count++
            left++
            right--
        } else if (sum < m) {
            left++
        } else {
            right--
        }
    }

    println(count)
}